src/math/p_not.c

```diff
@@ -0,0 +1,41 @@
+#include <pal.h>
+
+/**
+ *
+ * Element wise vector 'bitwise not' of input vector 'a'
+ *
+ * @param a     Pointer to input vector
+ *
+ * @param c     Pointer to output vector
+ *
+ * @param n     Size of 'a' and 'c' vector.
+ *
+ * @param p     Number of processor to use (task parallelism)
+ *
+ * @param team  Team to work with 
+ *
+ * @return      None
+ *
+ */
+
+#define GEN_FUNC(NAME,TYPE) \
+	/** NAME TYPE */ \
+	void NAME(const TYPE * restrict a, TYPE * restrict c, int n) \
+	{ \
+    	*c = ~*(a);  \
+        for (;--n;) \
+    	    *c = ~*(a + n); \
```

Comment:
@mateunho Thanks, fixed.

@aolofsson Yes, loops are very shallow ( will look into the possibility of using a duff's device to do some unrolling, or if -funroll-all-loops can be used without pigging out on size). Also I noticed it can be counterproductive to have a decreasing index. 

What I think we should do now is to define a number of platforms we want to target and the compiler flags we want to use ( O2 or O3, and more importantly on x86 the SSE flags. Or should we split x86 in 2 ). 

What is the preferred method of implementing platform specific code? #ifdef tree?